src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ContigAligner.java

```diff
@@ -54,56 +48,38 @@ public ContigAligner(final String referenceFilePath) throws IOException {
                 final String contigId = contigInfo._1.toString();
                 final byte[] sequence = contigInfo._2.toString().getBytes();
                 final AlnRgn[] alnRgns = bwaAlignSequence(bwaMem, contigId, sequence);
 
                 log.info("alnRgns : " + (alnRgns == null ? "null" : alnRgns.length));
-                // todo: parse cigar for internal indels
-                if (alnRgns.length > 1) {
-
-                    // filter out secondary alignments, convert to AlignmentRegion objects and sort by alignment start pos
-                    final List<AlignmentRegion> alignmentRegionList = Arrays.stream(alnRgns)
-                            .filter(a -> a.getSecondary() < 0)
-                            .map(AlignmentRegion::new)
-                            .sorted(Comparator.comparing(a -> a.startInAssembledContig))
-                            .collect(arrayListCollector(alnRgns.length));
-
-
-                    final Iterator<AlignmentRegion> iterator = alignmentRegionList.iterator();
-                    if ( iterator.hasNext() ) {
-                        AlignmentRegion current = iterator.next();
-                        while ( iterator.hasNext() ) {
-                            final AlignmentRegion previous = current;
-                            current = iterator.next();
-                            String homology = "NA";
-                            if (previous.endInAssembledContig >= current.startInAssembledContig) {
-                                homology = new String(Arrays.copyOfRange(sequence, current.startInAssembledContig - 1, previous.endInAssembledContig));
-                            }
-
-                            String insertedSequence = "NA";
-                            if (previous.endInAssembledContig < current.startInAssembledContig - 1) {
-                                insertedSequence = new String(Arrays.copyOfRange(sequence, previous.endInAssembledContig, current.startInAssembledContig));
-                            }
-                            final AssembledBreakpoint assembledBreakpoint = new AssembledBreakpoint(contigId, previous, current, homology, insertedSequence);
-                            assembledBreakpoints.add(assembledBreakpoint);
-                        }
-                    }
-                }
+                // filter out secondary alignments, convert to AlignmentRegion objects and sort by alignment start pos
+                final List<AlignmentRegion> alignmentRegionList = Arrays.stream(alnRgns)
+                        .filter(a -> a.getSecondary() < 0)
+                        .map(a -> new AlignmentRegion(assemblyId, contigId, a))
+                        .sorted(Comparator.comparing(a -> a.startInAssembledContig))
+                        .collect(SVUtils.arrayListCollector(alnRgns.length));
+                 alignedContigs.addAll(alignmentRegionList);
```

Comment:
rather than collecting into a new list and then adding that list to the final list, your final pipeline step could just stuff the new ones directly into the final list:
.forEach(alignedContigs::add);